website/docs/r/managed_application.html.markdown

```diff
@@ -0,0 +1,119 @@
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  lock_level          = "ReadOnly"
+  package_file_uri    = "https://github.com/Azure/azure-managedapp-samples/raw/master/Managed Application Sample Packages/201-managed-storage-account/managedstorage.zip"
+  display_name        = "TestManagedAppDefinition"
+  description         = "Test Managed App Definition"
+
+  authorization {
+    service_principal_id = data.azurerm_client_config.current.object_id
+    role_definition_id   = split("/", data.azurerm_role_definition.builtin.id)[length(split("/", data.azurerm_role_definition.builtin.id)) - 1]
+  }
+}
+
+resource "azurerm_managed_application" "example" {
+  name                       = "example-managedapplication"
+  location                   = azurerm_resource_group.example.location
+  resource_group_name        = azurerm_resource_group.example.name
+  kind                       = "ServiceCatalog"
+  target_resource_group_name = "infrastructureGroup"
+  application_definition_id  = azurerm_managed_application_definition.example.id
+
+  parameters = {
+    location                 = azurerm_resource_group.example.location
+    storageAccountNamePrefix = "storeNamePrefix"
+    storageAccountType       = "Standard_LRS"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Managed Application. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Managed Application should exist. Changing this forces a new resource to be created.
+
+* `location` - (Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created.
+
+* `kind` - (Required) The kind of the managed application. Possible values are `MarketPlace` and `ServiceCatalog`.
+
+* `target_resource_group_name` - (Required) The name of the target resource group which includes resources managed by Managed Application.
+
+* `application_definition_id` - (Optional) The fully qualified path of managed application definition ID.
+
+* `parameters` - (Optional) The name and value pairs that define the managed application parameters.
+
+* `plan` - (Optional) One `plan` block as defined below.
+
+* `tags` - (Optional) A mapping of tags to assign to the resource.
+
+---
+
+The `plan` block exports the following:
+
+* `name` - (Required) Specifies the name of the plan from the marketplace.
+
+* `product` - (Required) Specifies the product of the plan from the marketplace.
+
+* `publisher` - (Required) Specifies the publisher of the plan.
+
+* `version` - (Required) Specifies the version of the plan from the marketplace.
+
+* `promotion_code` - (Optional) Specifies the promotion code of the plan. 
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `id` - The ID of the Managed Application.
```

Comment:
I believe it would be convenient to expose the app's outputs as well. Do you think we could add that from the beginning?

Reply:
According by kt's comments previously, suggests to not add outputs from the beginning. Suggests to add them until user asks. Thus we don't need to maintain the outputs user doesn't need whereas outputs is easily to be added.

Reply:
What are the app outputs you would like to see @seiffert and in what instance would they be sueful?

Reply:
Sorry for the late response. The app outputs can be anything the app publisher decides to output,  so this would need to be a generic key/value map. 

Outputs could be URLs to a management interface deployed as part of the app or also just configuration values for clients to use the app. E.g. if you would deploy a managed app that contains a database server, the server name could be an output of the app. By making it an attribute of the resource, the server name could be used to provision the databases in that server.

Reply:
Of course, this  functionality can also be developed in a future PR. I was just looking at this for a potential use case of our team where outputs would be good to have.

Reply:
ahh thanks for explaining that makes a lot of sense, @neil-yechenwei could we get an `managed_app_outputs` map as suggested?

Reply:
@katbyte , may I ask you mean I need to add a Computed map attribute `managed_app_outputs` which includes `name\resource_group_name\kind\managed_resource_group_name\application_definition_id\parameters\plan\tags` in resource file, right? If yes, could we add a data source file to output these properties?

Reply:
Added